exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java

```diff
@@ -24,6 +24,7 @@
 public class TraceExporter implements SpanExporter {
 
   private final TraceServiceClient traceServiceClient;
+  private final MockTraceServiceClient mockTraceServiceClient;
```

Comment:
`TraceExporter` shouldn't know anything about mocking.
If you want to be able to replace `TraceServiceClient` with `MockTraceServiceClient`, you need to define an interface abstraction and implement it for mock and for production (wrapper around TraceServiceClient).

e.g.
```java
public interface CloudTraceClient {
  void batchWriteSpans(ProjectName name, List<Span> spans)
}

public class MockCloudTraceClient implements CloudTraceClient {
  ...
}

public class CloudTraceClientImpl implements CloudTraceClient {
  private final TraceServiceClient traceServiceClient;
  ...

  public final void batchWriteSpans(ProjectName name, List<Span> spans) {
    traceServiceClient.batchWriteSpans(name, spans);
  }
}
```

Does it make sense?

Reply:
Yep good idea, I've made the change. Please let me know if this was not what you were describing.